.github/scripts/validate-typespec-specs.ts

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env tsx
+/* eslint-disable no-console */
+
+import { readdir, stat } from 'fs/promises';
+import { join } from 'path';
+import { spawn } from 'child_process';
+
+async function findTspConfigDirectories(startDir: string): Promise<string[]> {
+  const directories: string[] = [];
+  
+  async function searchDirectory(dir: string): Promise<void> {
+    try {
+      const entries = await readdir(dir);
+      
+      // Check if this directory has a tspconfig.yaml
+      if (entries.includes('tspconfig.yaml')) {
+        directories.push(dir);
+        return; // Don't search subdirectories if we found a tspconfig.yaml
+      }
+      
+      // Search subdirectories
+      for (const entry of entries) {
+        const fullPath = join(dir, entry);
+        try {
+          const stats = await stat(fullPath);
+          if (stats.isDirectory() && !entry.startsWith('.') && entry !== 'node_modules') {
+            await searchDirectory(fullPath);
+          }
+        } catch (error) {
+          // Skip directories we can't access
+          continue;
+        }
+      }
+    } catch (error) {
+      // Skip directories we can't access
+      return;
+    }
+  }
+  
+  await searchDirectory(startDir);
+  return directories;
+}
+
+async function runTspCompile(directory: string): Promise<{ success: boolean; output: string }> {
```

Comment:
can you update so that multiple compile are run in parallel(starting with 4 create a const at the top to easily configure it).

Make sure the output is still buffered and logged when the command is done